### PR TITLE
Chore/small fixes

### PR DESCRIPTION
**Pull request by IceTDrinker**
### Resolves:

closes https://github.com/zama-ai/concrete-core-internal/issues/284

### Description

Some unchecled serialization ffi function were performing checks, I missed those in review (sorry), this PR fixes that and also changes some multiplications in ggsw encryption to wrapping multiplications

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [x] The tests on AWS have been launched and are successful (apply the `aws_test` to the PR to launch the tests on AWS)
* [x] The draft release description has been updated
* [x] Check for breaking changes and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: concrete-core/src/commons/crypto/secret/glwe.rs [hunk 1]
```diff
@@ -929,10 +929,11 @@ where
             .par_level_matrix_iter_mut()
             .zip(generators)
             .for_each(move |(mut matrix, mut generator)| {
-                let decomposition = encoded.0
-                    * (Scalar::ONE
+                let decomposition = encoded.0.wrapping_mul(
```
Comment (agnesLeroy) [inline concrete-core/src/commons/crypto/secret/glwe.rs:932]: Is it really a wrapping multiplication we should do here? Actually it shouldn't change anything if the decomposition is done properly no?

Comment (IceTDrinker) [reply concrete-core/src/commons/crypto/secret/glwe.rs:932]: I know I had some tests run in debug that crashed (as they check for overflows), other ggsw encryption already have it and Orel fixed one in the original seeded PR

Comment (IceTDrinker) [reply concrete-core/src/commons/crypto/secret/glwe.rs:932]: plus my current understanding is as we do things modularly we should (?) use wrapping ops ?

Comment (agnesLeroy) [reply concrete-core/src/commons/crypto/secret/glwe.rs:932]: Oh yes the wrapping mul is done in the sequential version but not here!

**Review by agnesLeroy — APPROVED** [review]

**Comment by agnesLeroy** [comment]
Looks good to me! Thank you!

**Review by aPere3 — APPROVED** [review]
Looks good thanks @IceTDrinker !

**Pull request merged**